### PR TITLE
Add reset options to "hiway-v1".

### PR DESCRIPTION
**Pull request by Gamenot**
See CHANGELOG.md

### CODE REVIEW

File: smarts/env/gymnasium/hiway_env_v1.py [hunk 1]
```diff
@@ -497,3 +504,12 @@ def scenario(self) -> Scenario:
...
         return self._smarts.scenario
+
+    @property
+    def smarts(self):
+        """Gives access to the underlying simulator. Use this carefully.
+
+        Returns:
+            smarts.core.smarts.SMARTS: The smarts simulator instance.
+        """
+        return self._smarts
```
Comment (Adaickalavan) [inline smarts/env/gymnasium/hiway_env_v1.py:515]: Considering we have previously provided `def scenario()`, what is the use case for additionally providing `def smarts()`?

File: smarts/env/tests/test_hiway_env_v1.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    agent_ids = set(agent_interfaces)
+    env: HiWayEnvV1 = gym.make(
+        "smarts.env:hiway-v1",
+        scenarios=["scenarios/sumo/loop"],
+        agent_interfaces=agent_interfaces,
+        action_options="unformatted",
+        observation_options="unformatted",
+        headless=True,
+        visdom=False,
+        fixed_timestep_sec=0.01,
```
Comment (Adaickalavan) [inline smarts/env/tests/test_hiway_env_v1.py:48]: Maybe we should simply make the `fixed_timestep_sec=0.01` as the default value in `HiWayEnvV1`.

File: smarts/env/tests/test_hiway_env_v1.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        MAX_EPISODES - 1
+    ), "Simulation must cycle through to the final episode."
+
+
+def test_hiway_env_v1_reset_with_scenario(env: HiWayEnvV1):
+    scenarios = ["scenarios/sumo/loop"]
+    scenario: Scenario = next(Scenario.scenario_variations(scenarios, [AGENT_ID]))
+
+    env.reset(options={"scenario": scenario, "start_time": 100})
+    assert "loop" in env.scenario.root_filepath
```
Comment (Adaickalavan) [inline smarts/env/tests/test_hiway_env_v1.py:98]: Given the env originally only had one scenario, namely `scenarios/sumo/loop`, testing reset functionality by resetting to the same scenario might be an insufficient test. We could try resetting the env to a different scenario to check whether the functionality works.

File: smarts/env/tests/test_hiway_env_v1.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    assert isinstance(env.unwrapped, HiWayEnvV1)
+    assert not (agent_ids - set(env.agent_interfaces))
+    matching_items = [
+        env.agent_interfaces[k] == agent_interfaces[k]
+        for k in env.agent_interfaces
+        if k in agent_interfaces
+    ]
+    assert all(matching_items)
+    assert len(env.agent_interfaces) == len(agent_interfaces)
+    assert not (agent_ids - env.agent_ids)
```
Comment (Adaickalavan) [inline smarts/env/tests/test_hiway_env_v1.py:60]: Wondering whether these checks could be simplified.

File: smarts/env/gymnasium/hiway_env_v1.py (hunk 1 above)
Comment (Gamenot) [reply smarts/env/gymnasium/hiway_env_v1.py:515]: Generally just for direct access to SMARTS, I am somewhat cautious about allowing this but if we do not allow it there may be little exposure to actually using SMARTS directly.

File: smarts/env/tests/test_hiway_env_v1.py (hunk 2 above)
Comment (Gamenot) [reply smarts/env/tests/test_hiway_env_v1.py:48]: Why `0.01`? The fixed time-step for the environment is `0.1` if not already given.

File: smarts/env/tests/test_hiway_env_v1.py (hunk 3 above)
Comment (Gamenot) [reply smarts/env/tests/test_hiway_env_v1.py:98]: Right I was grabbing something quick, I will try "cloverleaf".

File: smarts/env/tests/test_hiway_env_v1.py (hunk 4 above)
Comment (Gamenot) [reply smarts/env/tests/test_hiway_env_v1.py:60]: I also note that these checks should not be in the fixture. I am moving and simplifying them.

File: smarts/env/tests/test_hiway_env_v1.py (hunk 3 above)
Comment (Gamenot) [reply smarts/env/tests/test_hiway_env_v1.py]: Used `figure_eight`.

**Review by saulfield — APPROVED** [review]

**Pull request merged**